frontend/src/pages/surveys/components/preferences-step/preferences-step.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button, Input } from '#libs/components/components.js';
+import {
+  useAppForm,
+  useCallback,
+  useFormController,
+} from '#libs/hooks/hooks.js';
+import {
+  getSurveyCategories,
+  type SurveyInputDto,
+  surveyInputValidationSchema,
+} from '#packages/survey/survey.js';
+import {
+  DEFAULT_SURVEY_PAYLOAD,
+  INVALID_ARRAY_INDEX,
+  PREFERENCES_CATEGORIES,
+  SPLICE_COUNT,
+  TEXTAREA_MAX_LENGTH,
+  TEXTAREA_ROWS_COUNT,
+} from '#pages/surveys/libs/constants.js';
+
+import { SurveyCategory } from '../survey-category/survey-category.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  onSubmit: (options: string[]) => void;
+};
+
+const PreferencesStep: React.FC<Properties> = ({ onSubmit }) => {
+  const { control, errors, isValid, handleSubmit } = useAppForm<SurveyInputDto>(
+    {
+      defaultValues: DEFAULT_SURVEY_PAYLOAD,
+      validationSchema: surveyInputValidationSchema,
+    },
+  );
+  const {
+    field: { onChange: onCategoryChange, value: categoriesValue },
+  } = useFormController({
+    name: 'preferences',
+    control,
+  });
+
+  const hasOther = categoriesValue.includes('Other');
+
+  const handleFieldChange = useCallback(
+    (option: string) => {
+      const index = categoriesValue.indexOf(option);
+
+      if (index === INVALID_ARRAY_INDEX) {
+        onCategoryChange([...categoriesValue, option]);
+
+        return;
+      }
+
+      categoriesValue.splice(index, SPLICE_COUNT);
```

Comment:
Bad practise to mutate, just create a copy of array

Reply:
we can also use `toSpliced` method if it is supported

Reply:
`toSplices` isn't supported

Reply:
@v0ldemar01 I used slice instead, hope it will be acceptable. Also can change it back to filter

Reply:
Good choice